merlin/models/xgb/__init__.py

```diff
@@ -0,0 +1,145 @@
+from typing import List, Optional
+
+import nvtabular as nvt
+import pandas as pd
+import xgboost as xgb
+
+from merlin.io import Dataset
+from merlin.schema import Tags
+
+
+class XGBoost:
+    """
+    The class adapts an XGBoost model to work with the high level merlin-models API.
+
+    Example usage::
+
+        # get the movielens dataset
+        from merlin.datasets.entertainment import get_movielens
+
+        train, valid = get_movielens()
+
+        # Train an XGBoost model
+        from merlin.schema import Tags
+        from merlin.models.xgb import XGBoost
+
+        model = XGBoost("binary:logistic")
+        model.fit(train)
+
+        model.evaluate(valid)
+    """
+
+    def __init__(self, objective: str, *args, **kwargs):
+        """
+
+        List of XGBoost objective functions:
+        https://xgboost.readthedocs.io/en/stable/gpu/index.html#objective-functions
+        """
+        self.objective = objective
+
+    def fit(self, train: Dataset, params: Optional[dict] = None, **kwargs) -> xgb.Booster:
+        """Trains the XGBoost Model
+
+        Will use the columns tagged with the target_type passed to the
+        constructor as the labels.  And all other non-list columns as
+        input features.
+        """
+        objective = self.objective
+        target_tag = get_target_tag(objective)
+        self.target_columns = get_targets(train, target_tag)
+
+        # if the target is a regression, normalize values
+        if objective == "reg:logistic":
+            train = normalize_regession_targets(train)
```

Comment:
Do we want to do this by default? This would also mean we would re-normalize if the targets are already between 0 and 1.

Reply:
Removed